Library/Homebrew/formula_creator.rb

```diff
@@ -9,11 +9,21 @@ module Homebrew
   #
   # @api private
   class FormulaCreator
-    attr_reader :args, :url, :sha256, :desc, :homepage
-    attr_accessor :name, :version, :tap, :mode, :license
-
-    def initialize(args)
-      @args = args
+    attr_reader :url, :tap, :sha256, :desc, :homepage
+    attr_accessor :name
+
+    sig {
+      params(name: T.nilable(String), version: T.nilable(String), tap: T.nilable(String), mode: T.nilable(Symbol),
+             license: T.nilable(String), fetch: T::Boolean, head: T::Boolean).void
+    }
+    def initialize(name, version, tap:, mode:, license:, fetch: true, head: false)
+      @name = name
+      @version = Version.new(version) if version
+      @tap = Tap.fetch(tap || "homebrew/core")
```

Comment:
```suggestion
      @tap = Tap.fetch(tap || "homebrew/core")
      raise TapUnavailableError, @tap.name unless @tap.installed?

```

Reply:
I moved the check to `fc.verify` method to keep constructor simple - no exception throwing and other complicated logic.